enhancements/network/ovn-on-hypershift.md

```diff
@@ -0,0 +1,275 @@
+
+#### OVN Node components
+**ovn-controller**: itself will not require any changes as it appears to already have support for the new functionality required. It must support configuration of the SB (southbound) database connection as a URL instead of an IP address. Further, when using TLS, it must set SNI (Server Name Indication) so that the management cluster is able to route the TLS connection to the correct database instance.
+
+**OVN-Kubernetes node:** should not require any changes.
+
+#### Node-DB connectivity
+By (ab)using the TLS Passthrough feature of OpenShift Routes, we can expose the southbound database directly to the client cluster.
+
+### Monitoring
+
+The control plane components, as pods in the management cluster, will need to be monitored by the management monitoring infrastructure. We will need to create a ServiceMonitor for each control plane in question.
+
+One complication is that we should create AlertManager alerts on a per-customer basis. That means, inevitably, that we will need some kind of management layer for these alerts. What form this takes is not yet defined.
+
+### Risks and Mitigations
+
+**OVSBD Raft and DNS** - the ovsdb team does not currently test Raft with dynamic member IPs. While the team believes this should work, it must be added to their test plan. Also, we have missed the window for any new changes to OVS for the 4.11 release. If any changes to ovsdb are needed, we either need exceptional processes or workarounds.
+
+### Open Questions 
+
+
+What storage / volume types will be available for the OVN DBs to use? Can we assume a functioning CSI environment? Will this storage have sufficient performance? Can we generate a reasonable size prediction? To answer this, we will need to utilize the existing Perf&Scale processes to characterize performance.
+
+What bandwidth will the sbdb traffic consume? Will the added latency cause any issues? Should we enable ovn-controller conditional monitoring? At the time this document was being drafted, the OVN team is re-examining the performance impacts of ovn-controller's conditional monitor feature.
+
+If the bandwidth or performance is not acceptable, will we have to deploy some kind of ovsdb proxy on the infra nodes? Assuming we cannot *require* infra nodes, will we nevertheless need to make that option available to customers?
+
+How do we report control-plane status -- which references "invisible" pods -- to the end-user? Does the end-user need to know that ovsdb is somehow degraded? What do we show in the `network` ClusterOperator object?
+
+How do we report control-plane status to the SRE teams? They will need to react to losses of quorum, etc. Can we deploy alerts? How do we handle cross-version alert definitions?
+
+How do we detect guest cluster MTU? We cannot just probe from the CNO as normal. We will either need to require MTU to be configured, or execute a probe pod. We do, however, support changing MTU
```

Comment:
i assume this is modifiable via the CNO config today, and we will provide that as referenced config to HostedCluster?

/cc @csrwng

Reply:
My understanding is that mtu is configured in the `network.operator.openshift.io` resource. We currently don't let you specify that from HostedCluster. So far we've only exposed configuration resources in the `config.openshift.io` group, which includes `network.config.openshift.io`. We would need to add explicit support for the operator config and sync it to the guest cluster so that the cno can find it where expected.

Reply:
That's correct, it's assumed to be a "lower level" knob.

Another option is to probe for the MTU via a Job once the first guest Node appears. We would then persist this.

FWIW, end-users may still wish to modify the `network.operator.openshift.io` if they intend to use Multus or advanced OVN features (e.g. flow exporting, policy auditing).

Reply:
Added MTU probe job.